app/javascript/chat/components/ChannelButton.jsx

```diff
@@ -0,0 +1,123 @@
+import { h, Component, createRef } from 'preact';
+import PropTypes from 'prop-types';
+
+class ChannelButton extends Component {
+  buttonRef = createRef();
+
+  componentDidMount() {
+    const { isActiveChannel = false } = this.props;
+
+    if (isActiveChannel) {
+      this.buttonRef.current.click();
+    }
+  }
+
+  renderChannelImage = () => {
+    const { channel, newMessagesIndicator, discoverableChannel } = this.props;
+
+    return (
+      <span
+        data-channel-slug={channel.channel_modified_slug}
+        className={
+          discoverableChannel
+            ? 'chatchanneltabindicator'
+            : `chatchanneltabindicator chatchanneltabindicator--${newMessagesIndicator}`
+        }
+        data-channel-id={channel.chat_channel_id}
+      >
+        <img
+          src={channel.channel_image}
+          alt="pic"
+          className={
+            channel.channel_type === 'direct'
+              ? 'chatchanneltabindicatordirectimage'
+              : 'chatchanneltabindicatordirectimage invert-channel-image'
+          }
+        />
+      </span>
+    );
+  };
+
+  render() {
+    const {
+      channel,
+      handleSwitchChannel,
+      otherClassname,
+      newMessagesIndicator,
+      isUnopened,
+      discoverableChannel,
+      triggerActiveContent,
+    } = this.props;
+
+    return (
+      <button
```

Comment:
refactor (blocking): Use the design system [Button](https://storybook.forem.com/?path=/story/3-components-buttons--default) component.